addons/product_images/models/ir_cron_trigger.py

```diff
@@ -0,0 +1,26 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class ir_cron_trigger(models.Model):
+    _inherit = "ir.cron.trigger"
+
+    @api.constrains('cron_id')
+    def _check_cron_id(self):
+        """ Check if `ir_cron_fetch_image` is already triggered
+
+        Ensure that there is a maximum of one trigger at a time for `ir_cron_fetch_image`.
+
+        :return: None
+        :raises ValidationError: If there is already an `ir_cron_fetch_image` triggered
+        """
+        ir_cron_fetch_image_id = self.env.ref('product_images.ir_cron_fetch_image').id
+        if self.cron_id.id != ir_cron_fetch_image_id:
+            return
+        nb_cron = self.env['ir.cron.trigger'].search_count(
+            [('cron_id', '=', ir_cron_fetch_image_id)]
+        )
+        if nb_cron > 1:
+            raise ValidationError(_("You can't trigger this cron more than once at a time!"))
```

Comment:
Be me, a lambda user. I have a few products in my PoS for which I would like to download an image. I click the magic button "fetch images" to fetch the images. I go grab a coffee. When I come back I see that only 10 images have been downloaded out of my 50 products. I click the magic button again. "You can't trigger this cron more than once at a time!" says the huge warning in the middle of my screen. I google "trigger cron", read a few Wikipedia pages, and decide not to click the magic button ever again.

Reply:
As a lambda user who uses the wizard, you should never see this ValidationError because the wizard will return a UserError even before opening.
https://github.com/odoo/odoo/blob/f1403116cd49a65eacf5dec5358489473a4cdb8e/addons/product_images/wizard/product_fetch_image_wizard.py#L41-L47

Reply:
This is not likely but it could also happen that the same admin works with two tabs open or that two admins decide to fetch some images simultaneously. Said like this it sounds impossible but two admins could very well be working on the setup of a DB in parallel and regularly fetch images as they list new products.

Reply:
I agree with anv in saying that you are leaking implementation details, you should explain the technical reason *why* there cannot be 2 triggers at a time for this cron instead of explaining *how* you implemented it.

Be a developer who tries to use your model, for some reason he is trying to re-trigger the cron via code. The contrains forbid such an operation and there is a traceback pointing the function. He reads it, doesn't understand *why* it cannot be triggered twice because the comment just *explains* the code without given more informations, he then blames the code to find the missing information in your commit message but it is missing too.

Even if I specified and implemented the trigger API, I had difficulties to understand the reason why it cannot be triggered twice and we ended up talking about the implementation for one hour at work.

Doing reviews is hard, when someone points you what he thinks is a problem but *he is wrong* then you should ask yourself *why did he thought it was wrong ?*. Maybe it is a communication problem, docstrings / comments / error messages / commit messages do help a lot when it comes to communicate important design decision ;)

Reply:
In that case, no error will be returned because I followed @Julien00859 advice to check for the cron before trigger it in the wizard. 
https://github.com/odoo/odoo/blob/f1403116cd49a65eacf5dec5358489473a4cdb8e/addons/product_images/wizard/product_fetch_image_wizard.py#L106-L111

Reply:
I improved the comment sections in `product_images/models/ir_cron_trigger.py` and `product_images/data/ir_cron_data.xml`